src/components/YearPicker.vue

```diff
@@ -0,0 +1,62 @@
+<template>
+<div class="container outer">
+  <div class="container box">
+    <p class="year">{{ this.year }}</p>
+    <button class="btn btn-primary float-left" @click="onClick($event, mode=true)">
+      &lt; {{ this.prevYear }}
+    </button>
+    <button class="btn btn-primary float-right" @click="onClick($event, mode=false)">
+      {{ this.nextYear }} &gt;
+    </button>
+  </div>
+</div>
+</template>
+
+<script lang="ts">
+import { Component, Vue } from 'vue-property-decorator';
+
+@Component
+export default class YearPicker extends Vue {
+  public year: number = new Date().getFullYear();
+
+  public prevYear: number = this.year - 1;
```

Comment:
Rather than these being explicit properties (that are additional states), change it what they really are - just getters based on the only state required - the currentYear

you might need to read up on Vue/Vuex getters for this

this further reduce our click handler to just one line - adding the number coming in